packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -623,3 +627,31 @@ class RunCommand extends RunCommandBase {
     );
   }
 }
+Future<void> generateVersionFile(FlutterProject flutterProject) async {
```

Comment:
There is already code in the FlutterProject class to parse all of this information, why are you repeating the process yourself?

Reply:
didn't know . I m using it now in the new commit